registry/machine.go

```diff
@@ -102,5 +165,16 @@ func (r *EtcdRegistry) RemoveMachineState(machID string) error {
 	if etcd.IsKeyNotFound(err) {
 		err = nil
 	}
+
+	// Delete metadata
```

Comment:
Could this not be inferred by the `Delete` action against the machine's `metadata` key? This comment is unnecessary.